Sources/SwiftProtobuf/SelectiveVisitor.swift

```diff
@@ -0,0 +1,268 @@
+// Sources/SwiftProtobuf/SelectiveVisitor.swift - Base for custom Visitors
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// A base for Visitors that only expect a subset of things to called.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+/// A base for Visitors that only expects a subset of things to called.
+internal protocol SelectiveVisitor: Visitor {
+  // Adds nothing.
+}
+
+/// Default impls for everything so things using this only have to write the
+/// methods they expect.  Asserts to catch developer errors, but becomes
+/// nothing in release to keep code size small.
+///
+/// NOTE: This is an impl for *everything*. This means the default impls
+/// provided by Visitor to bridge packed->repeated, repeated->singular, etc
+/// won't kick in.
+internal extension SelectiveVisitor {
```

Comment:
yea, I didn't want to use _Base_ as it isn't something most should need.  _Minimal_, _SubSet_, etc. all didn't seem to work.